core/src/main/java/com/linecorp/armeria/common/CommonPools.java

```diff
@@ -32,27 +29,17 @@
  */
 public final class CommonPools {
 
-    private static final ScheduledExecutorService BLOCKING_TASK_EXECUTOR;
-    private static final EventLoopGroup WORKER_GROUP;
-
-    static {
-        // Threads spawned as needed and reused, with a 60s timeout and unbounded work queue.
-        final ScheduledThreadPoolExecutor scheduledThreadPoolExecutor = new ScheduledThreadPoolExecutor(
-                Flags.numCommonBlockingTaskThreads(),
-                ThreadFactories.newThreadFactory("armeria-common-blocking-tasks", true));
-        scheduledThreadPoolExecutor.setKeepAliveTime(60, TimeUnit.SECONDS);
-        scheduledThreadPoolExecutor.allowCoreThreadTimeOut(true);
-        BLOCKING_TASK_EXECUTOR = scheduledThreadPoolExecutor;
-
-        WORKER_GROUP = EventLoopGroups.newEventLoopGroup(Flags.numCommonWorkers(),
-                                                         "armeria-common-worker", true);
-    }
+    // Threads spawned as needed and reused, with a 60s timeout and unbounded work queue.
+    private static final BlockingTaskExecutor BLOCKING_TASK_EXECUTOR =
+            BlockingTaskExecutor.builder().threadNamePrefix("armeria-common-blocking-tasks").build();
+    private static final EventLoopGroup WORKER_GROUP =
+            EventLoopGroups.newEventLoopGroup(Flags.numCommonWorkers(), "armeria-common-worker", true);
 
     /**
-     * Returns the default common blocking task {@link ScheduledExecutorService} which is used for
+     * Returns the default common blocking task {@link BlockingTaskExecutor} which is used for
      * potentially long-running tasks which may block I/O threads.
      */
-    public static ScheduledExecutorService blockingTaskExecutor() {
+    public static BlockingTaskExecutor blockingTaskExecutor() {
```

Comment:
This would be a breaking change. Revert the return type?

Reply:
Hmm, only this would be sufficient ?

Is it okay to leave as current change in here ? https://github.com/line/armeria/pull/3389/files/a340ec6b5f4e02686c31b8145042506706af5a76#diff-87779ed185d85af477c7a62b37787bc167631bb162be02de5bf0a11c1e8dbf09R33

Reply:
Yes, we don't have to change for the private field.
Only `public` and `protected` could be APIs.
JFYI, https://wiki.eclipse.org/Evolving_Java-based_APIs_2